nixos/modules/services/games/archisteamfarm.nix

```diff
@@ -27,6 +27,9 @@ let
       PasswordFormat = 4;
       Enabled = c.enabled;
     });
+
+    # TODO: drop with 24.11
+    stateDirectory = if lib.versionAtLeast config.system.stateVersion "24.05" then "/var/lib/archisteamfarm" else "/var/lib/asf";
```

Comment:
Why should this be dropped with 24.11?
After all the recommendation is for stateVersion to be not changed and the removal would break things.

Reply:
> Why should this be dropped with 24.11?

Because I don't want to carry the backwards compatibility forever and I can't test if it continues to work.

>  After all the recommendation is for stateVersion to be not changed and the removal would break things.

Good point, I changed it, that people should set the stateDirectory option.